Jakarta Bean Validation 2.0

gunnarmorling opened the pull request:
## Specification PR template
When creating a specification project release review, create two PRs with the content divided as follows.

Include the following in PR#1:
- [x] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of wombat_x.y.pdf
- [x] Specification HTML in the form of wombat_x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [x] ~~For a Progress Review, that sufficient progress has been made on
      a Compatible Implementation and TCK, to ensure that the spec is
      implementable and testable.~~
- [x] For a Release Review, a summary that a Compatible Implementation
      is complete, passes the TCK, and that the TCK includes sufficient
      coverage of the specification. The TCK users guide MUST include
      the instructions to run the compatible implementations used to
      validate the release.  Instructions MAY be by reference.
  - [x] Compatible implementation: see https://github.com/eclipse-ee4j/beanvalidation-spec/issues/246
  - [x] TCK includes sufficient coverage: See the TCK [coverage report](https://ci.eclipse.org/bean-validation/job/EE4J-TCK-Release/lastBuild/org.hibernate.beanvalidation.tck$beanvalidation-tck-tests/artifact/org.hibernate.beanvalidation.tck/beanvalidation-tck-tests/2.0.5/beanvalidation-tck-tests-2.0.5-coverage.html)
  - [x] TCK instructions: See the [user guide](http://docs.jboss.org/hibernate/beanvalidation/tck/2.0/reference/html_single/)
- [x] The URL of the OSSRH staging repository for the api, javadoc:
      https://oss.sonatype.org/content/repositories/staging/jakarta/validation/jakarta.validation-api/2.0.2/
- [x] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:\
http://download.eclipse.org/ee4j/bean-validation/beanvalidation-tck-dist-2.0.5.zip
http://download.eclipse.org/ee4j/bean-validation/beanvalidation-tck-dist-2.0.5.tar.gz
- [x] The URL of the compatibility certification request issue:
      https://github.com/eclipse-ee4j/beanvalidation-spec/issues/246

Include the following in PR#2:
- [x] Specification JavaDoc in the wombat/x.y/apidocs directory.

gunnarmorling commented:
This PR as well as the linked PR for the API docs and the compatibility certification request issue are good for a first review from my side now.

gunnarmorling commented:
Hey @kwsutter, there's nothing to add from my side right now, so I think we could remove the "draft" label?

ivargrimstad commented:
> Hey @kwsutter, there's nothing to add from my side right now, so I think we could remove the "draft" label?

According to the process discussed in yesterday's spec committee meeting, the "draft" label will be removed by someone from the spec committee when "pre reviews" have been done.

gunnarmorling commented:
OK, that's cool. Wasn't quite sure whether the label was there due to me
marking the PR as "wip" initially. Thx!

starksm64 commented:
There is a typo in the specs where they indicate where a copy of the EFSL can be found:
(Copy of the EFPL license here)

should be:
(Copy of the EFSL license here)

I have released a new version of the spec build here:
https://ci.eclipse.org/bean-validation/job/EE4J-spec/

bshannon commented on bean-validation/2.0/_index.md:
```diff
@@ -0,0 +1,35 @@
+* [Jakarta Bean Validation 2.0 Specification Document](./bean-validation_2.0.pdf) (PDF)
+* [Jakarta Bean Validation 2.0 Specification Document](./bean-validation_2.0.html) (HTML)
+* [Jakarta Bean Validation 2.0 Javadoc](./apidocs)
+* [Jakarta Bean Validation 2.0 TCK](http://downloads.eclipse.org/jakarta/bean-validation/2.0.5/beanvalidation-tck-dist-2.0.5.zip)
```
The URL should be:
http://downloads.eclipse.org/jakarta/bean-validation/2.0/beanvalidation-tck-dist-2.0.5.tar.gz
(unless this is converted to a zip file)
Note directory name is just "2.0".

starksm64 replied:
Both zip and tar.gz files exist in the downloads area. The PR has been updated to use the zip version.

pzygielo replied:
Is `downloads.eclipse.org` eclipse-internal or geo-restricted service? I get `NXDOMAIN` from DNS.

gunnarmorling replied:
Ok, I'm confused :) So atm. the file is here: http://download.eclipse.org/ee4j/bean-validation/beanvalidation-tck-dist-2.0.5.zip (i.e. no version info in the path, only in the file name)

Is this the correct location? If so, I'll update the PR to link there.

gunnarmorling replied:
Updated as per that URL.

bshannon replied:
The URL in the _index.md file needs to be the URL **where the TCK will be published when it is final**.
Obviously the TCK hasn't been published there yet.

And my URL above was wrong, it should be "download", not "downloads":
http://download.eclipse.org/jakarta/bean-validation/2.0/beanvalidation-tck-dist-2.0.5.zip

gunnarmorling replied:
Ok, updated as per @bshannon's last comment.

bshannon commented on bean-validation/2.0/_index.md:
```diff
@@ -0,0 +1,35 @@
+* [Jakarta Bean Validation 2.0 Javadoc](./apidocs)
+* [Jakarta Bean Validation 2.0 TCK](http://downloads.eclipse.org/jakarta/bean-validation/2.0.5/beanvalidation-tck-dist-2.0.5.zip)
+* Maven coordinates
+  * [jakarta.validation:jakarta.validation-api:jar:2.0.2](https://search.maven.org/artifact/jakarta.validation/jakarta.validation-api/2.0.2/jar)
```
"2.0.2" -> "2.0.5"

gunnarmorling replied:
No, the API version is 2.0.2 indeed; 2.0.5 is the TCK version.

bshannon replied:
Oops, you're right.

gunnarmorling commented:
> I have released a new version of the spec build here:
> https://ci.eclipse.org/bean-validation/job/EE4J-spec/

@starksm64, the new version is missing the "Status: Final" bit:

```
<pre>Specification: Jakarta Bean Validation
 
-Version: 2.0.2
+Version: 2.0
 
-Status: Final
+Status: 
 
-Release: 2019-08-01</pre>
+Release: 2019-08-05</pre>
```

gunnarmorling commented:
> spec document references "Java API for JavaBean validation", need to remove Java references

@bshannon, so to be sure, will this work: "This document is the specification of the Jakarta API for Bean Validation in Jakarta EE and Java SE"?

gunnarmorling commented:
Pushed a commit to add the _index.md file and fix the TCK link.

bshannon commented:
> > spec document references "Java API for JavaBean validation", need to remove Java references
> 
> @bshannon, so to be sure, will this work: "This document is the specification of the Jakarta API for Bean Validation in Jakarta EE and Java SE"?

I would say:

This is the Jakarta Bean Validation specification for use with Jakarta EE and Java SE.

Or something like that.  (Don't need the "API for".)

gunnarmorling commented:
I've reworked and force-pushed my last commit, it's addressing all of your remarks now, @bshannon. This one is good to go from my PoV, I have nothing to add unless further review feedback indicates so.

starksm64 commented:
# Spec Review Checklist

1. Spec PR
  - [x] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [x] Directory of form {spec}/x.y
  - [x] PDF of form {spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [x] HTML of form {spec}-spec-x.y.html ("-spec" preferred but not required)
  - [x] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [x] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [x] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [x] Staging repository link of the form https://oss.sonatype.org/conte/repositories/staging/jakarta/{spec}/jakarta-{spec}-api/x.y.z/
  - [x] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [x] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [x] Second PR for just apidocs

2. _index.md
  - [x] Link to spec pdf
  - [x] Link to spec html
  - [x] Link to apidocs
  - [x] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip
  - [x] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [x] Name of and link to at least one Compatible Implementation

3. javadocs
  - [x] Footer contains Eclipse copyright and link to license
  - [x] ESFL license is included, usually as doc-files/speclicense.html
  - [x] no META-INF directory in PR
  - [x] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [x] Correct spec title
  - [x] Version number of the form x.y, not x.y.z
  - [x] Correct Eclipse copyright line
  - [x] No DRAFT or SNAPSHOT

5. Spec HTML
  - [x] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [x] EFTL license file, preferably named LICENSE.md (has EFTL.txt)
  - [x] User's Guide (or equivalent documentation) (docs/reference/html_single/index.html)
  - [x] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [x] Software requirements listed
  - [x] Installation and configuration described
  - [x] How to run tests
  - [x] Where to file challenges

8. Compatibility certification request
  - [x] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [x] SHA-256 fingerprint matches staged TCK zip file
ebab3232311439dfc93559ca0dfa8cc230f51ab221cdc0a4901a8533f129f3ad
  - [x] Request issue has `certification` label.

9. TCK results summary
  - [x] Page is hosted by Compatible Implementation project
  - [x] Includes all information from certification request
  - [x] Summary includes number of tests passed, failed, errors

starksm64 commented:
> The specification document uses the Duke logo, which may not be used due to trademark issues,
> as described in [this IPZilla entry](http://dev.eclipse.org/ipzilla/show_bug.cgi?id=16051).
> Please change it to the Jakarta EE logo.

I have updated the spec and given it to @gunnarmorling to add to the PR.

gunnarmorling commented:
The spec document has been updated to contain the shooner image.

starksm64 commented:
Added draft label back in until Gunnar reruns the CI against the latest TCK dist that was updated for GlassFish 5.1 testing issues.

gunnarmorling commented:
I've re-run that latest TCK build against Hibernate Validator 6.0.17.Final and updated the related information in the specification certifaction request https://github.com/eclipse-ee4j/beanvalidation-spec/issues/246.

starksm64 commented:
This is good to go for vote now. Verified that the cert request and TCK results have been updated to match the current TCK dist.

starksm64 commented:
@dblevins the api jars have been released to maven central

starksm64 commented:
- [x] The designated specification committee representative adds this final checklist to the main PR.
 - [ ] The specification committee representative merges the specification PRs.
 - [ ] The designated specification committee representative calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job.
 - [ ] The specification project member who created the staging release promotes the specification api jars to maven central. An example release job script can be found here https://wiki.eclipse.org/MavenReleaseScript.
 - [ ] The EMO updates the specification page with the ballot results.
This list goes on the committed spec index page.
 - [ ] The specification project team should go through the merged spec website page to verify all the links are valid.
 - [ ] The specification project team should approve the compatibility request.
 - [ ] The compatible implementation project/vendor MUST send an email tck@eclipse.org for approval of the compatible implementation for trademark usage.
 - [ ] The specification project team should merge any final release branch as appropriate for the branch management for the project.

dblevins commented:
## PROMOTED FILES



- https://download.eclipse.org/jakartaee/bean-validation/2.0/beanvalidation-tck-dist-2.0.5.tar.gz
- `05afb564efaa6d893dbeb32fb614d6e2bd18f09cf76244386f3b377d725b14b8`

```
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2.0.22 (GNU/Linux)

iQIcBAABAgAGBQJdcXnaAAoJEKvgVyXnfCtEEPkQAKeCUWjkjfnY/YFawH0OQZBP
oCHQEr+6UBNE+NbVEUTaitEsaC/D52p02ULDWvPqKST460TIC+bQzjyr2FLpqKRD
bCb/+VxR/sEi3xeRuAioEanz5M6bGsOyOxiBd+UHuFFe8KQX2PYwQQwDpF2jGYhP
80MdNrfKhAZ91Epzz+pFyMj4HXpsGZlRuOayrXF7qsghtSBumlY4M8GlxGr+eIL2
ahsmID6iWm8a6QUon087rYBP66bInFypj9GLCz/Baa//Vjd/nYQXG+MmIGak8AJT
lZg//cq2ThtiTGRAweeYPEzB7CwQ1Z80Dr//NbW7dg2Z6y2GnZ7RMNE7JbYD7PDG
/tWBiZ7oLhG7kuwi8yENMQxej8WC/3jaZa8SGC/wPwkV7hFLUrIUvUMQ4OlxPikE
HJZHLAI8FVbzeDYfRoqE1QCgMRQlV/YrEF6Tk2VZGwUSPw+Cv0+bNyPYIIDHoNTK
ITAGWqdig8q4yKLtWyDnYPjqttNZ6CC6JLrmQYU11WYSBlwQ8H+iTicdZVIbAM3h
4v/tzE3Qj0QKRTVupneGhG++OcAfzaClEErjhTnVKZDbXFNnIyzZhtHdctWCRF/J
zqL64u+UyMGuHlbQjATbJ0Gpv3JcM9weU7Yq7X5Dn2SZsJCOHkH0NYxsIRrNSqKq
+5lxeGMseEvjGWdf9cnN
=btGY
-----END PGP SIGNATURE-----
```

- https://download.eclipse.org/jakartaee/bean-validation/2.0/beanvalidation-tck-dist-2.0.5.zip
- `b6778914f78bfcce5d6934347d71502603b1b0a6bbfdfbcf956271c367d40974`
```
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2.0.22 (GNU/Linux)

iQIcBAABAgAGBQJdcXnaAAoJEKvgVyXnfCtE7ocQAMow8pcbmgKBLUFa2XBAzbw0
6oY/734xbxLNYQ8Ri5EzG6jMRGQUaDrcj5M13uhFJebOcAvazjaVUCTeoC27V2kG
GK3KCczF+VN318a07wBOObgs2+Dz6oThxmQVuZYSE9QT7sERPhPLC7BUMuCXQI0c
vIs5ygzVgYE7Ij0dX752ZGiUuIPGUPHv9ZOVLy9YYFl97r1JqxsT2Qr0M8Qrg9Ni
1X13lQG9d4wZXKwWvyHfXImd1RD8/8+I/t7FiObM7T6QAuFjJlo1uxQAX4BmgmOi
CFpxtm30ZUld2daWu/wTdCEC++j9d72IvcP+IXIfpDf5vYf82j9ex5TcCgSVSQKY
6tOIQUWQS7PprnT+ADRZStbnlONTel8oIc6tioMDae8qgDz5gx6NzUVYmZVsSN93
CJ0+cAzf8TXjCkjCqiqsJyiAw0+yI81eKJlszZD/YdIcwfwkqjphvWUXsOTR2lPL
/RWkXJdcRt5v3ox8I25EJqCkE91u+UonwJqS4wJeTXcw0KzcC3WwIOtrfGmf+01c
5NiXcrOf9mFSXp5iTGCgzxXicJKx8Fiv3BkzOsNv2DBAPveB0iDzbgk3MTX6Csjb
nqDSE3d0cLgyucs4mMTcRdCdK0fVcqfmD7sgvuJpLH9ClTiERs2dUf+Ui7QAaqOV
S/Kz63MwZJIuf4BEf4VT
=2sgx
-----END PGP SIGNATURE-----
```

gunnarmorling commented:
Yeehaw, thanks a lot, everybody!
